cli/flox-rust-sdk/src/models/env_registry.rs

```diff
@@ -204,14 +240,47 @@ pub struct RegisteredEnv {
     pub pointer: EnvironmentPointer,
 }
 
+/// PID of an environment's activation.
+#[derive(Debug, Clone, Serialize, Deserialize, Hash, Eq, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct Pid(u32);
```

Comment:
> question: Why are we rolling our own `Pid` type rather than taking or wrapping `unistd::Pid`?

Can you describe the advantage? Is it to address the type casting? Looks like we'd still need to implement `is_running()` on top of it.

Reply:
we are using this type only really by converting its data back into a `unistd::Pid`.

I think a newtype with particular semantics is a reasonable convention.
Going from one representation into a another equivalent representation whose methods all require an upcast back into the former is odd.

Here it seems using `unistd::Pid` might mess with the De/Serialize derives (and probably proptest). That would make it less unconventional, since you probably dont want to provide custom impls for all of those for a simple utility like this.
If that's the case i think at least motivating this in a comment would be enough.

Reply:
Done in https://github.com/flox/flox/pull/1837/commits/23c92da2e794f421fa5cad280e14d808f27f005a